WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Common/Navigation/MigrationFlowCoordinator.swift

```diff
@@ -3,16 +3,61 @@ import UserNotifications
 
 /// Coordinator for the migration to jetpack flow
 final class MigrationFlowCoordinator: ObservableObject {
-    // beware that changes won't be published on the main thread,
+
+    // MARK: - Dependencies
+
+    private let migrationEmailService: MigrationEmailService?
+    private let userPersistentRepository: UserPersistentRepository
+
+    // MARK: - Properties
+
+    // Beware that changes won't be published on the main thread,
     // so always make sure to return to the main thread for UI updates
     // related to this property.
     @Published private(set) var currentStep = MigrationStep.welcome
 
+    // MARK: - Init
+
+    init(migrationEmailService: MigrationEmailService? = try? .init(),
+         userPersistentRepository: UserPersistentRepository = UserPersistentStoreFactory.instance()) {
+        self.migrationEmailService = migrationEmailService
+        self.userPersistentRepository = userPersistentRepository
+        self.userPersistentRepository.jetpackContentMigrationState = .inProgress
+    }
+
+    deinit {
+        if userPersistentRepository.jetpackContentMigrationState != .completed {
+            self.userPersistentRepository.jetpackContentMigrationState = .notStarted
+        }
+    }
```

Comment:
The `init` and the `de-init` of `MigrationFlowCoordinator` updates the content migration state. When the migration starts, we mark the state as `inProgress`, and when it ends, we mark the state back to `notStarted` if the migration didn't complete.